RELEASES.md

```diff
@@ -1,3 +1,142 @@
+Version 1.52.0 (2021-04-29)
+============================
+
+Language
+--------
+- [Added the `unsafe_op_in_unsafe_fn` lint, which checks whether the unsafe code
+  in an `unsafe fn` is wrapped in a `unsafe` block.][79208] By default this lint
+  is allow by default, and may become a warning or hard error in a
+  future release.
+- [You can now cast mutable references to arrays to a pointer of the same type as
+  the element.][81479]
+
+Compiler
+--------
+- [Upgraded the default LLVM to LLVM 12.][81451]
+
+Added tier 3\* support for the following targets.
+
+- [`s390x-unknown-linux-musl`][82166]
+- [`riscv32gc-unknown-linux-musl` & `riscv64gc-unknown-linux-musl`][82202]
+- [`powerpc-unknown-openbsd`][82733]
+
+Libraries
+---------
+- [`OsString` now implements `Extend` and `FromIterator`.][82121]
+- [`cmp::Reverse` now has `#[repr(transparent)]` representation.][81879]
+- [`Arc<impl Error>` now implements `error::Error`.][80553]
+- [All stable methods on integers are now `const`.][80962]
+
+Stabilised APIs
+-------------
+- [`Arguments::as_str`]
+- [`Peekable::next_if_eq`]
+- [`Peekable::next_if`]
+- [`char::MAX`]
+- [`char::REPLACEMENT_CHARACTER`]
+- [`char::UNICODE_VERSION`]
+- [`char::decode_utf16`]
+- [`char::from_digit`]
+- [`char::from_u32_unchecked`]
+- [`char::from_u32`]
+- [`slice::partition_point`]
+- [`str::rsplit_once`]
+- [`str::split_once`]
+
+The following previously stable APIs are now `const`.
+
+- [`char::len_utf8`]
+- [`char::len_utf16`]
+- [`char::to_ascii_uppercase`]
+- [`char::to_ascii_lowercase`]
+- [`char::eq_ignore_ascii_case`]
+- [`u8::to_ascii_uppercase`]
+- [`u8::to_ascii_lowercase`]
+- [`u8::eq_ignore_ascii_case`]
+
+Cargo
+-----------------------
+- [`cargo update` now works with the `--offline` flag.][cargo/9279]
```

Comment:
I generate it with this [tool]

Reply:
Ah, I see. Yea, there is lag between things landing in cargo and getting merged to rust-lang/rust, so just using dates may catch some extra PRs.

I have a script for bumping versions and prepping the cargo changelog. It looks at the submodules in rust-lang/rust using a bunch of git commands to find the correct range of changes (it looks roughly like [this](https://github.com/ehuss/cargo-new-release/blob/36fd9dd2fe0bedcd9bdd29d3a1ecc06187f4e80b/src/main.rs#L183-L202)).  Looking at submodules is probably not appropriate for your tool since it looks like it is a little more general purpose.

I've been thinking that it might be nice to have the bot that sets milestones to also propagate those to the cargo repository. I'm not sure if that might help here.

Another thought is that instead of `use-parent-for-timeframe`, it could look at a specific branch ([rust-1.52.0](https://github.com/rust-lang/cargo/commits/rust-1.52.0)). That would require knowing/updating the version each time, though.